server/src/main/java/org/elasticsearch/cluster/metadata/MetadataIndexTemplateService.java

```diff
@@ -617,6 +607,38 @@ public ClusterState addIndexTemplateV2(
         return ClusterState.builder(currentState).metadata(Metadata.builder(currentState.metadata()).put(name, finalIndexTemplate)).build();
     }
 
+    public Map<String, List<String>> v2TemplateOverlaps(
```

Comment:
Shall we document this method and since it's now called explicitly maybe add a test in `MetadataIndexTemplateServiceTests` ?

Reply:
Great suggestion, let me do that!